DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/matmul/src/matmul_demo.cpp

```diff
@@ -0,0 +1,196 @@
+#include <iomanip>
+#include <iostream>
+
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+#include "matmul.hpp"
+
+// Fills a matrix with random numbers within the range [lbound, ubound).
+void FillRand(float *M_matrix, int lbound, int ubound, int elements) {
+  for (int element = 0; element < elements; element++) {
+    M_matrix[element] =
+        static_cast<float>(rand()) /
+            (static_cast<float>((RAND_MAX) / (ubound - lbound))) +
+        lbound;
+  }
+}
+
+// Compares two matrices; returns true iff they are equal.
+bool EqualMat(float *C_matrix, float *C_reference, int rows, int cols,
+              int kNumMatrices) {
+  int matsize = rows * cols;
+  bool passed = true;
+  constexpr float kEpsilon = 0.01f;  // floating-point error threshold value
+
+  for (int matrix_idx = 0; matrix_idx < kNumMatrices; matrix_idx++) {
+    for (int col = 0; col < cols; col++) {
+      for (int row = 0; row < rows; row++) {
+        int idx = matrix_idx * matsize + col * rows + row;
+        if (abs(C_matrix[idx] - C_reference[idx]) > kEpsilon) {
+          passed = false;
+#if DEBUG == 1
+          std::cout << "Error: C[" << col << "][" << row << "] = "
+                    << C_matrix[idx] 
+                    << " but REF[" << col << "][" << row << "] = "
+                    << C_reference[idx] << std::endl;
+#endif
+        }
+        if (!std::isfinite(C_matrix[idx])) {
+          passed = false;
+#if DEBUG == 1
+          std::cout << "C[" << col << "][" << row << "] = " << C_matrix[idx]
+                    << " is not finite" << std::endl;
+#endif
+        }
+      }
+    }
+  }
+  return passed;
+}
+
+// Output a matrix to the screen (assumes column-major format).
+void PrintMat(float *M_matrix, int rows, int cols) {
+  for (int row = 0; row < rows; row++) {
+    for (int col = 0; col < cols; col++) {
+      std::cout << std::fixed;
+      std::cout << std::setprecision(2);
+      std::cout << std::setw(8) << M_matrix[col * rows + row] << " ";
+    }
+    std::cout << std::endl;
+  }
+}
+
+// Transpose given matrices in m and store the result in m_transposed.
+void TransposeMat(float *M_matrix, float *M_transposed, int rows, int cols,
+                  int kNumMatrices) {
```

Comment:
k_num_matrices

Reply:
for the num matrices variable (and all other functions in the testbench that have it), I've updated it to be `num_matrices` to be consistent with the other arguments (without the `k`). let me know if I should change this

Reply:
Yes that is good, the `k` should only be used for constants, which `num_matrices` is not :)